src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,297 @@
+    ),
+  ]?.length;
+
+  // all is selected if there is nothing selected(for empty selected) and not selected is 0 (for all selected)
+  const notSelectedCount = [
+    ...new Set(
+      options?.filter((item: any) => !item?.selected)?.map((item: any) => item?.id)
+    ),
+  ]?.length;
+
+  const optionsDisplayed = searchTerm
+    ? (options || []).filter(
+        (item: any) => item?.text?.toLowerCase()?.includes(searchTerm?.toLowerCase())
+      )
+    : options || [];
+
+  return (
+    <ClickAwayListener
+      onClickAway={(e) => {
+        // Check if the clicked element is the anchorEl, if so, prevent the clickaway logic
+        if (popRef?.current && popRef?.current?.contains(e?.target)) {
+          return;
+        }
+
+        // Your regular clickaway logic here
+        setShowPopover(false);
+      }}
+    >
+      <Box>
+        <Input
+          data-testid={dataTestId}
+          data-cy={dataCy}
+          error={error}
+          onClick={() => {
+            setShowPopover(true);
+          }}
+          endAdornment={
+            <Box
+              cs={{
+                display: "inline-flex",
+                gap: "spacing.xs",
+                alignItems: "center",
+                paddingLeft: "spacing.sm",
+                background: "white",
+              }}
+            >
+              {hasSelectedCount ? (
+                !notSelectedCount ? (
+                  // eslint-disable-next-line
+                  <Chip priority="low" role="primary" size="md">
+                    ALL
+                  </Chip>
+                ) : selectedCount ? (
+                  // eslint-disable-next-line
+                  <Chip priority="low" role="primary" size="md">
+                    {selectedCount}
+                  </Chip>
+                ) : null
+              ) : null}
+              <SvgIcon
+                icon={faChevronDown}
+                onClick={() => {
+                  setShowPopover(true);
+                }}
+              />
+            </Box>
+          }
+          fullWidth
+          ref={popRef}
+          label={label}
+          placeholder={placeholder}
+          value={
+            multipleSelect
+              ? selectedCount
+                ? label
+                : ""
+              : options?.find((item: any) => item?.id === defaultValue)?.text || ""
+          }
+          readOnly={true}
+        />
+        {showPopover ? (
+          <Popover
+            isBlock
+            anchorEl={popRef}
+            placement="top-start"
+            modifiers={[
+              {
+                name: "preventOverflow",
+                options: {
+                  mainAxis: false, // true by default
+                },
+              },
+            ]}
+            style={{
+              padding: "0px",
+              zIndex: 9,
+              width: "calc(100% - 40px)",
+              maxWidth: maxWidth ? maxWidth : "40%",
```

Comment:
Question; why do we set max width to 40% if not specified? Shouldn't it be default value/unset ?

Reply:
yes you're right this should be unset